### ISSUE TITLE
Duplicate Labels Removed

### ISSUE THREAD

**Issue by anidh** [ultralytics/ultralytics#14858]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

Hi There,

While training the yolov8 instance segmentation model. For various files I get the duplicate labels removed. For one of the file let's say xyz.jpg The contents of the txt file is 
```
1 0.2625 0.7038 0.2784 0.7063 0.2796 0.7138 0.2664 0.7232 0.2656 0.7819 0.2247 0.8066 0.1651 0.806 0.1658 0.7026 0.2221 0.701
1 0.5058 0.5145 0.5007 0.6188 0.4786 0.6177 0.39 0.6116 0.39 0.5107 0.422 0.5121
1 0.6398 0.611 0.6383 0.6991 0.6403 0.7026 0.7247 0.7073 0.7485 0.7097 0.7529 0.7085 0.7548 0.7026 0.7572 0.695 0.756 0.6088 0.736 0.6128
1 0.6285 0.6147 0.6398 0.611 0.736 0.6128 0.756 0.6088 0.7579 0.5166 0.7093 0.5117 0.6326 0.5066 0.6285 0.6024
1 0.7579 0.5166 0.756 0.6088 0.7572 0.695 0.8222 0.6944 0.8234 0.5411 0.8241 0.5182 0.7791 0.5166
1 0.8893 0.5241 0.8881 0.6967 0.8222 0.6944 0.8234 0.5411 0.8241 0.5182 0.8433 0.5211
1 0.8433 0.5211 0.8893 0.5241 0.9088 0.5217 0.9088 0.5111 0.9119 0.3552 0.845 0.3487
1 0.7791 0.5166 0.8241 0.5182 0.8433 0.5211 0.845 0.3487 0.7786 0.3408
1 0.7093 0.5117 0.7579 0.5166 0.7791 0.5166 0.7786 0.3408 0.7088 0.3381
1 0.7088 0.3381 0.7093 0.5117 0.6326 0.5066 0.6347 0.4107 0.6405 0.3322
1 0.8462 0.1609 0.8816 0.1578 0.9143 0.167 0.9119 0.3552 0.845 0.3487
1 0.78 0.1629 0.8462 0.1609 0.845 0.3487 0.7786 0.3408
1 0.7088 0.3381 0.7786 0.3408 0.78 0.1629 0.7711 0.1564 0.7117 0.157
1 0.6371 0.3075 0.6415 0.2231 0.6422 0.1798 0.6448 0.1554 0.6506 0.1468 0.7117 0.157 0.7088 0.3381 0.6405 0.3322
1 0.6371 0.3075 0.6326 0.24 0.6275 0.218 0.614 0.1994 0.5096 0.2114 0.5084 0.2963 0.5118 0.3151 0.6095 0.3042
1 0.6347 0.4107 0.6405 0.3322 0.6371 0.3075 0.6095 0.3042 0.5118 0.3151 0.5082 0.4152 0.6073 0.4099
1 0.6326 0.5066 0.6347 0.4107 0.6073 0.4099 0.5082 0.4152 0.5058 0.5145 0.5897 0.5101
1 0.6068 0.6198 0.6285 0.6147 0.6285 0.6024 0.6326 0.5066 0.5897 0.5101 0.5058 0.5145 0.5007 0.6188 0.5885 0.6179
1 0.5 0.7161 0.5657 0.7222 0.614 0.7271 0.6275 0.7187 0.6318 0.6208 0.6068 0.6198 0.5885 0.6179 0.5007 0.6188
1 0.5 0.7161 0.5101 0.7222 0.4981 0.835 0.4718 0.8258 0.3871 0.8154 0.3888 0.7673 0.3871 0.7144 0.486 0.7201
1 0.5007 0.6188 0.5 0.7161 0.486 0.7201 0.3871 0.7144 0.3826 0.7097 0.39 0.6116 0.4786 0.6177
1 0.5101 0.7222 0.6227 0.7395 0.6342 0.7426 0.6309 0.8125 0.6287 0.8339 0.616 0.8541 0.6066 0.8531 0.4981 0.835
1 0.1687 0.2592 0.1668 0.2282 0.1696 0.1798 0.1749 0.16 0.1898 0.1609 0.2077 0.16 0.2173 0.1572 0.2293 0.1537 0.2365 0.1566 0.2839 0.1841 0.2854 0.197 0.268 0.1947 0.2628 0.2763 0.2288 0.2663 0.2012 0.2606
1 0.2628 0.2763 0.2632 0.2947 0.2702 0.2955 0.2885 0.2947 0.3633 0.3102 0.3898 0.311 0.3883 0.2253 0.3883 0.2086 0.3761 0.2061 0.2854 0.197 0.268 0.1947
1 0.3883 0.2086 0.4143 0.2131 0.4463 0.2114 0.4661 0.2063 0.5096 0.2114 0.5084 0.2963 0.5118 0.3151 0.5034 0.31 0.3898 0.311 0.3883 0.2253
1 0.5118 0.3151 0.5082 0.4152 0.4892 0.4152 0.3874 0.4126 0.3908 0.3346 0.3898 0.311 0.5034 0.31
1 0.3405 0.4079 0.2914 0.395 0.2637 0.3832 0.2702 0.2955 0.3114 0.2947 0.3633 0.3102 0.3898 0.311 0.3908 0.3346 0.3874 0.4126 0.3708 0.4132
1 0.3405 0.4079 0.2914 0.395 0.2637 0.3832 0.2702 0.2955 0.3114 0.2947 0.3633 0.3102 0.3898 0.311 0.3908 0.3346 0.3874 0.4126 0.3708 0.4132
1 0.2702 0.2955 0.2637 0.3832 0.2262 0.3789 0.1908 0.3797 0.172 0.3789 0.1687 0.3265 0.1687 0.2792 0.1687 0.2592 0.2012 0.2606 0.2288 0.2663 0.2628 0.2763 0.2632 0.2947
1 0.2904 0.4079 0.2914 0.395 0.2637 0.3832 0.2262 0.3789 0.1908 0.3797 0.172 0.3789 0.1694 0.3936 0.1658 0.4287 0.1622 0.4768 0.1636 0.4819 0.166 0.4895 0.224 0.4919 0.2637 0.4895 0.263 0.4087
1 0.3874 0.4126 0.39 0.5107 0.3657 0.5113 0.2659 0.5125 0.2637 0.4895 0.263 0.4087 0.3405 0.4079 0.3708 0.4132
1 0.5058 0.5145 0.422 0.5121 0.39 0.5107 0.3874 0.4126 0.4892 0.4152 0.5082 0.4152
1 0.39 0.6116 0.39 0.5107 0.3657 0.5113 0.2659 0.5125 0.2683 0.602 0.3544 0.6086
1 0.2659 0.5125 0.2683 0.602 0.2283 0.5888 0.1648 0.5918 0.1668 0.498 0.166 0.4895 0.224 0.4919 0.2637 0.4895
1 0.1658 0.7026 0.1648 0.5918 0.2283 0.5888 0.2683 0.602 0.2625 0.7038 0.2221 0.701
1 0.3826 0.7097 0.39 0.6116 0.3544 0.6086 0.2683 0.602 0.2625 0.7038 0.2784 0.7063 0.2796 0.7138
1 0.2656 0.7819 0.268 0.8195 0.3361 0.8201 0.3871 0.8154 0.3888 0.7673 0.3871 0.7144 0.3826 0.7097 0.2796 0.7138 0.2664 0.7232
0 0.9981 0.0595 0.9737 0.9972 0 0.9632 0.0169 0

```
It says for this file there were 24 duplicate labels removed. Can you help me which how this logic works? The image contains lot of overlapped object in a box. The labels for the object class is 1.

### Environment

Ultralytics YOLOv8.2.51 🚀 Python-3.12.4 torch-2.3.1 CUDA:0 (NVIDIA RTX A6000, 48685MiB)
Setup complete ✅ (64 CPUs, 251.5 GB RAM, 3432.6/3666.0 GB disk)

OS                  Linux-6.0.12-76060012-generic-x86_64-with-glibc2.31
Environment         Linux
Python              3.12.4
Install             pip
RAM                 251.53 GB
CPU                 AMD Ryzen Threadripper PRO 3975WX 32-Cores
CUDA                11.8

### Minimal Reproducible Example

NA

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by pderrenger** [comment]
@anidh thank you for your detailed report. The message "duplicate labels removed" indicates that the training process has detected and removed overlapping or duplicate labels to ensure the integrity of the training data. This typically happens when multiple annotations for the same object are too close to each other, which can confuse the model during training.

To understand this better, you can check if the issue persists with the latest version of the Ultralytics package. If it does, please ensure your annotations are accurate and do not overlap excessively. If you continue to encounter this issue, feel free to submit a PR or provide further details, and we can investigate this further.

For more information on common issues and troubleshooting, you can visit our [YOLO Common Issues Guide](https://docs.ultralytics.com/guides/yolo-common-issues/).

**Comment by Y-T-G** [comment]
It's based on whether the boxes are the same. The segmentation is turned into a box.

https://github.com/ultralytics/ultralytics/blob/db4c43bafbed1e1298c4436b8c7e7c6e5b9f3eec/ultralytics/data/utils.py#L144
